test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Server/OrderLinesController.cs

```diff
@@ -0,0 +1,63 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="OrderLinesController.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.OData.Formatter;
+using Microsoft.AspNetCore.OData.Query;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+
+namespace Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Server
+{
+    public class OrderLinesController : ODataController
+    {
+        public OrderLinesController()
+        {
+        }
+
+        [EnableQuery]
+        public IActionResult Get()
+        {
+            var orderLines = CommonEndToEndDataSource.OrderLines;
+
+            if (orderLines == null)
+            {
+                return NotFound();
+            }

```

Comment:
We don't need to return not found for `GET /entityset` requests, we should just return an empty collection if there are not items. Under what circumstances would the `CommonEndToEndDataSource.OrderLines` property be null?

Reply:
I'll update this in the next PR.